### PR TITLE
Drop attachment requirements for media download

### LINKED ISSUE #645: `TestMediaFilenames` incorrectly tests Content-Disposition
`Content-Disposition` is not a spec defined behaviour (it's not mentioned in https://spec.matrix.org/v1.7/client-server-api/ other than for the filename, and complement should not be prescribing that servers act a certain way without a spec to cover it.

It's up to homeservers to decide what media types may be presented inline or as attachments (and something we're trying to fix in https://github.com/matrix-org/synapse/pull/15988 to fix a unsatisfactory situation).

### PR DESCRIPTION
**Pull request by Half-Shot**
Fixes #645 
For https://github.com/matrix-org/synapse/pull/15988

### CODE REVIEW

File: tests/media_filename_test.go [hunk 1]
```diff
@@ -119,7 +122,7 @@ func TestMediaFilenames(t *testing.T) {
 }
 
 // Returns content disposition information like (mediatype, filename)
```
Comment (erikjohnston) [inline tests/media_filename_test.go:124]: Update the docstring please?

File: tests/media_filename_test.go [hunk 2]
```diff
@@ -139,15 +142,16 @@ func downloadForFilename(t *testing.T, c *client.CSAPI, mxcUri string, diffName
...
-	if filename, ok := params["filename"]; ok {
-		return filename
-	} else {
-		t.Fatalf("Content Disposition did not have filename")
-
-		return ""
+	if mediaType != "inline" {
+		t.Fatalf("Found unexpected mediatype %s, expected attachment", mediaType)
 	}
+	return "", false
```
[the remark this answers is not in the document]
Comment (Half-Shot) [reply tests/media_filename_test.go:156]: The implications I got from https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition was that it didn't, but it was just a cursory read. 

https://www.rfc-editor.org/rfc/rfc6266#section-4.2 / 4.3 imply they can have filenames, so I'll fix the test.

**Comment by Half-Shot** [comment]
@kegsay has been review requested because of the SVG file, but I'm unsure how much he cares to review it. Of course, more than welcome :)

File: tests/media_filename_test.go [hunk 3]
```diff
@@ -138,16 +160,17 @@ func downloadForFilename(t *testing.T, c *client.CSAPI, mxcUri string, diffName
...
-
-	if mediaType != "attachment" {
-		t.Fatalf("Found unexpected mediatype %s, expected attachment", mediaType)
+	filename, hasFilename := params["filename"]
+	if mediaType == "attachment" {
+		if hasFilename {
+			return filename, true
+		} else {
+			t.Fatalf("Content Disposition did not have filename")
+			return "", true
```
Comment (clokep) [inline tests/media_filename_test.go:169]: Isn't there no need to return here after `t.Fatalf`?

File: tests/media_filename_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (clokep) [reply tests/media_filename_test.go]: It looks like we're still asserting that the filenames only exist for attachments though. Not sure if we want to update it or not.

**Review by erikjohnston — COMMENTED** [review]
Otherwise i think LGTM

File: tests/media_filename_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (itsrachelfish) [reply tests/media_filename_test.go]: Inline files still have filenames - this is what determines what the default filename is when you "Save As"

File: tests/media_filename_test.go (hunk 3 above)
Comment (kegsay) [reply tests/media_filename_test.go:169]: Correct as it panics so control flow stops. It doesn't hurt from a readability PoV though.

**Comment by clokep** [comment]
I updated this to handle the remaining review comments. I think this is ready for another review.

**Review by erikjohnston — APPROVED** [review]

**Pull request merged**